resources/views/dockerfile.blade.php

```diff
@@ -17,10 +17,20 @@
     && mkdir -p storage/logs \
     && php artisan optimize:clear \
     && chown -R www-data:www-data /var/www/html \
-    && sed -i 's/protected \$proxies/protected \$proxies = "*"/g' app/Http/Middleware/TrustProxies.php \
     && echo "MAILTO=\"\"\n* * * * * www-data /usr/bin/php /var/www/html/artisan schedule:run" > /etc/cron.d/laravel; \
     if [ -d .fly ]; then cp .fly/entrypoint.sh /entrypoint; chmod +x /entrypoint; fi;
 
+# Different syntax for Laravel 11 and previous versions 
+# For setting TrustProxies
+@if( preg_match("~\^|^1[1-9]~", $laravel_version ) )
+RUN sed -i='' '/->withMiddleware(function (Middleware \$middleware) {/a\
+        \$middleware->trustProxies(at: "*");\
+' bootstrap/app.php;
+@else
+RUN sed -i 's/protected \$proxies/protected \$proxies = "*"/g' app/Http/Middleware/TrustProxies.php
+@endif
```

Comment:
The original `sed` was a part of the previous RUN step.  I presume that it was split out because dealing with the bash if statement was messy.  That's not a problem using blade `@if` statements.  My suggestion is to replace the two `RUN` statements with `&&`, add a `\` to the end of these lines, and move this entire block back to line 20.

Reply:
trying this out!